src/palace/manager/api/admin/config.py

```diff
@@ -24,38 +26,20 @@ class AdminClientFeatureFlags(ServiceConfigurationWithLimitedEnvOverride):
         description="Show QuickSight dashboards only for sysadmins.",
     )
 
-    # The following fields CANNOT be overridden by environment variables.
-    # Setting `const=True` ensures that the default value is not overridden.
-    # Add them to the one of the `environment_override_*` Config settings
-    # below to prevent them from being overridden.
-    # NB: Overriding the `env_prefix` with `env=...` here may lead to
-    #     incorrect values in warnings and exceptions, since `env_prefix`
-    #     is used to generate the full environment variable name.
-    enable_auto_list: bool = Field(
+    # The following fields CANNOT be overridden by environment variables, so
+    # their types are set to Literal[True] to prevent them from being overridden.
+    enable_auto_list: Literal[True] = Field(
         True,
-        const=True,
         description="Enable auto-list of items.",
     )
-    show_circ_events_download: bool = Field(
+    show_circ_events_download: Literal[True] = Field(
         True,
-        const=True,
         description="Show download button for Circulation Events.",
     )
-
-    class Config:
-        env_prefix = "PALACE_ADMINUI_FEATURE_"
-
-        # We use lower camel case aliases, since we're sending to the web.
-        alias_generator = _snake_to_camel_case
-
-        # Add any fields that should not be overridden by environment variables here.
-        # - environment_override_warning_fields: warnings and ignore environment
-        # - environment_override_error_fields: raise exception
-        environment_override_warning_fields: set[str] = {
-            "enable_auto_list",
-            "show_circ_events_download",
-        }
-        environment_override_error_fields: set[str] = set()
+    model_config = SettingsConfigDict(
+        env_prefix="PALACE_ADMINUI_FEATURE_",
+        alias_generator=AliasGenerator(serialization_alias=to_camel),
```

Comment:
Pydantic has its own alias generator `to_camel` that I switched this to use, and removed `_snake_to_camel_case`.